### DISCUSSION TITLE
Experimental roaster ac heater connection?

### DISCUSSION THREAD

**Discussion by russ999** [artisan-roaster-scope/artisan#825] [Q&A]
What’s the easiest way to connect the heating element of a new experimental roaster to artisan?  I have built a popper style roaster.   Obviously it’s not a supported machine in artisan. I have already connected 4 TCs using a center 304 device via usb and I can see the temperature plots on artisan on my MacBook.  Now I just need to connect the heater and control it from artisan (ideally pid/profile eventually). The heater is a 120v ac resistive wire heater, at the moment controlled manually with a pwm knob. Motor control is not required. 

I gather there are many ways to do this but I’m hoping you can recommend the fastest, simplest, easies, cheapest methods so I can focus more on coffee and less on soldering, rewiring,  debugging, etc. Thanks!

**Comment by poundy** [comment]
if you're only using PWM to control your heating, then a PWM output phidget is quick and relatively easy. Alternatively, you can use the "open" design aspects of TC4 and use the PID in Artisan to control that.  Check out GreenCardigan and TC4 on github (https://github.com/greencardigan/TC4-shield) for some code
